src/height_above_ground.cpp

```diff
@@ -0,0 +1,240 @@
+ *   (at your option) any later version.                                     *
+ *                                                                           *
+ ****************************************************************************/
+
+#include <iostream>
+#include <filesystem>
+#include <thread>
+
+#include <pdal/PipelineManager.hpp>
+#include <pdal/Stage.hpp>
+#include <pdal/util/ProgramArgs.hpp>
+#include <pdal/pdal_types.hpp>
+#include <pdal/Polygon.hpp>
+#include <pdal/PipelineWriter.hpp>
+
+#include <gdal_utils.h>
+
+#include "utils.hpp"
+#include "alg.hpp"
+#include "vpc.hpp"
+
+using namespace pdal;
+
+namespace fs = std::filesystem;
+
+
+void HeightAboveGround::addArgs()
+{
+    argOutput = &programArgs.add("output,o", "Output point cloud file", outputFile);
+    argOutputFormat = &programArgs.add("output-format", "Output format (las/laz/copc)", outputFormat);
+    argAlgorithm = &programArgs.add("algorithm", "Height Above Ground algorithm to use: nn (Nearest Neighbor) or delaunay (Delaunay).", algorithm, "nn");
+    argReplaceZWithHeightAboveGround = &programArgs.add("replace-z", "Replace Z dimension with height above ground (true/false).", replaceZWithHeightAboveGround, false);
+
+    // args - NN
+    argNNCount = &programArgs.add("nn-count", "The number of ground neighbors to consider when determining the height above ground for a non-ground point", nnCount, 1);
+    argNNMaxDistance = &programArgs.add("nn-max-distance", "Use only ground points within max_distance of non-ground point when performing neighbor interpolation.", nnMaxDistance, 0);
+
+    // args - Delaunay
+    argDelaunayCount = &programArgs.add("delaunay-count", "The number of ground neighbors to consider when determining the height above ground for a non-ground point.", delaunayCount, 10);
+}
+
+bool HeightAboveGround::checkArgs()
+{
+    if (!argOutput->set())
+    {
+        std::cerr << "missing output" << std::endl;
+        return false;
+    }
+
+    if (argOutputFormat->set())
+    {
+        if (outputFormat != "las" && outputFormat != "laz" && outputFormat != "copc")
+        {
+            std::cerr << "unknown output format: " << outputFormat << std::endl;
+            return false;
+        }
+    }
+    else
+        outputFormat = "las";  // uncompressed by default
+
+    if (!argAlgorithm->set())
+    {
+        std::cerr << "missing algorithm" << std::endl;
+        return false;
+    }
+    else
+    {
+        if (!(algorithm == "nn" || algorithm == "delaunay"))
+        {
+            std::cerr << "unknown algorithm: " << algorithm << std::endl;
+            return false;
+        }
+    }
+
+    if (!argReplaceZWithHeightAboveGround->set())
+    {
+        replaceZWithHeightAboveGround = false; // default
+    }
+
+    if (!argNNCount->set())
+    {
+        nnCount = 1; // default
+    }
+
+    if (!argNNMaxDistance->set())
+    {
+        nnMaxDistance = 0; // default
+    }
```

Comment:
aren't these defaults already defined in addArgs() ? do we need this code at all?